build/index.js

```diff
@@ -397,6 +378,14 @@ async function buildDocument(document, documentOptions = {}) {
 
   doc.modified = metadata.modified || null;
 
+  const root = document.fileInfo.root;
+  doc.source = {
+    hash: metadata.hash || null,
+    folder: document.fileInfo.folder,
+    repository_url: REPOSITORY_URLS[root],
+    branch: getCurrentGitBranch(root),
```

Comment:
Another thing, I think we should consider going back a bit and compute the relevant URLs here in Node instead of concocting URLs with the bits of data, in the client. 
Thing is, everything we stuff into `doc.source` will be included in every single `index.json` we build. And we build over 50k of them. I don't think we need to over-optimize this but I definitely think we should be a bit delicate. We shouldn't make a bunch of data available "just in case" it's needed. I'd rather we "work backwards" and make sure we only have the information necessary. 

Kinda lika Rest JSON vs. GraphQL, except we don't have a language to express it. Instead, we need to be accurately aware of what this `doc.source` is *used* for. And based on that, make sure it's only containing the information that's actually needed. Does that make sense?

Like, if you know you only need the `firstName` and `lastName`, we should make sure we *don't* do this:
```
doc.userInfo = {
  firstName: user.firstName,
  middleName: user.middleName,
  lastName: user.lastName,
}
```

Reply:
Done!